runtime/lua/vim/lsp/completion.lua

```diff
@@ -145,8 +145,8 @@ local function get_completion_word(item)
       --    label: insert
       --
       -- Typing `i` would remove the candidate because newText starts with `t`.
-      local text = item.insertText or item.textEdit.newText
-      return #text < #item.label and text or item.label
+      local text = parse_snippet(item.insertText or item.textEdit.newText)
+      return #text < #item.label and vim.fn.matchstr(text, '\\k*') or item.label
```

Comment:
I'm not sure about the matchstr here. Without it we'd use `INT16_C(c)` as word. With it `INT16_C`

My opinion is that the latter is better if you _don't_ expand the snippet. Because after selecting an entry you can continue typing `(whatever`. If `(c)` is inserted too you instead need to backtrack.

If you expand the snippet, it will insert `INT16_C(c)` with `c` being the tabstop/placeholder as expected.

Reply:
I agree that it's nice to not expand the snippet. Another argument if favour of that is that it might also be confusing to insert placeholders without starting a proper snippet session.